### PR TITLE
More flexible tokenizer

### PR DESCRIPTION
**Pull request by saarikoski-jules**
Adding working tokenizer to master. 
Tokenizer separates special characters from other input according to constant g_token_table located in token_table.h. Saves individual tokens to a linked list.

### CODE REVIEW

File: srcs/tokenizer.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+t_token *validate_tokens(char **input) 
+{
+	int		i;
+	t_token	*head;
+	t_token	*token;
+
+	i = 0;
+	head = NULL;
+	token = NULL;
```
Comment (LJmartin94) [inline srcs/tokenizer.c]: variable token is initialised but never used right?

File: srcs/tokenizer.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+	len = recognize_special_char(input, *j);
+	if (len > 0)
+	{
+		token = create_token(input, *k, *j - *k);
+		if (token != NULL)
+			add_token(head, token);
+		token = create_token(input, *j, len);
+		if (token != NULL)
+			add_token(head, token);
```
Comment (LJmartin94) [inline srcs/tokenizer.c:44]: Are you creating two tokens, one for the special character /separator, and one for the token that follows?

File: srcs/tokenizer.c (hunk 1 above)
Comment (saarikoski-jules) [reply srcs/tokenizer.c]: Oops!

File: srcs/tokenizer.c (hunk 2 above)
Comment (saarikoski-jules) [reply srcs/tokenizer.c:44]: One for specialchar and one for everything before from last copied part

**Review by LJmartin94 — CHANGES_REQUESTED** [review]
Just the one uninitialised variable we talked about, but looks amazing.

**Review by LJmartin94 — APPROVED** [review]

**Pull request merged**